### PR TITLE
BL- 488 Feature/mentor model update

### PR DESCRIPTION
**Pull request by willisbridges**
## Description

Co-authors: Jason Gersing, Cristian Ibarra-Villegas, Martha Gonzalez

Mentor collection update. Part of overarching plan to redo Database architecture to capture desired data with additional fields. For visibility, check this [excel spreadsheet](https://docs.google.com/spreadsheets/d/1dt54x0OVqgNfmGfW_omdQtEu6NXPfMZz_DUQKR7HCXo/edit?usp=sharing) (specifically the Refactor Final Draft tab) to see our high-level plans for refactoring each table.

Compiled new fields to Mentor class for architectural overhaul.

## Type of Change
- [X] New feature

## Checklist
- [X] My code follows PEP8 style guide
- [X] I have removed unnecessary print statements from my code
- [X] I have made corresponding changes to the documentation if necessary
- [X] My changes generate no errors
- [X] No commented-out code
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

LOOM:
https://www.loom.com/share/43c4ec60203b40feb7ddc535ee76a4e6

### CODE REVIEW

File: app/schema.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 class Mentor(BaseModel):
     profile_id: constr(max_length=255)
+    created_at: datetime
+    updated_ar: datetime
```
Comment (sayerjk) [inline app/schema.py:9]: typo right ar `updated_ar`

**Review by annakoduru6 — DISMISSED** [review]
Besides the typo Sayer pointed out (updated_at), looks good.

File: app/schema.py (hunk 1 above)
Comment (marthabg) [reply app/schema.py]: fixed

File: app/schema.py [hunk 2]
```diff
@@ -18,9 +20,11 @@ class Mentor(BaseModel):
     industry_knowledge: bool
     pair_programming: bool
     commitment: bool
-    referred_by: constr(max_length=255)
+    referred_by: Optional[constr(max_length=255)]
     other_info: Optional[constr(max_length=2500)]
-    validate_status: constr(max_length=255)
+    validate_status: Literal['approved', 'rejected', 'pending']
+    is_active: bool
```
Comment (sayerjk) [inline app/schema.py:26]: Have we discussed setting default values here?

File: app/schema.py [hunk 3]
```diff
@@ -18,9 +20,11 @@ class Mentor(BaseModel):
...
+    accepting_new_mentees: bool
```
Comment (sayerjk) [inline app/schema.py:28]: Have we considered setting defaults for `is_active` and `accepting_new_mentees`?

**Review by sspradling78 — DISMISSED** [review]
Looks good, I approve.

File: app/schema.py [hunk 4]
```diff
@@ -44,6 +50,8 @@ class MentorUpdate(BaseModel):
     referred_by: Optional[constr(max_length=255)]
     other_info: Optional[constr(max_length=2500)]
     validate_status: Optional[constr(max_length=255)]
```
Comment (zdravoj) [inline app/schema.py:52]: validate_status in MentorUpdate collection needs to be changed from a string to stringLiteral to reflect same changes in Mentor collection

File: app/schema.py [hunk 5]
```diff
@@ -18,16 +20,20 @@ class Mentor(BaseModel):
...
+    accepting_new_mentees: bool
 
     class Config:
         extra = Extra.forbid
 
 
 class MentorUpdate(BaseModel):
-    profile_id: Optional[constr(max_length=255)]
+    profile_id: constr(max_length=255)
+    created_at: Optional[datetime]
```
Comment (zdravoj) [inline app/schema.py:35]: created_at should not be an alterable field - it is set when the mentor is created. Please remove created_at from MentorUpdate collection.

**Review by zdravoj — CHANGES_REQUESTED** [review]
validate_status in MentorUpdate collection needs to be a stringLiteral with the same list of string choices as in Mentor collection.

created_at should not exist as an updatable option in MentorUpdate collection. This field is set when the mentor is created and cannot be altered.

**Review by zdravoj — APPROVED** [review]

**Review by opeterschmidt — APPROVED** [review]

**Comment by bummings** [comment]
> This PR should be connected to BL-488; BL-490 doesn't seem to exist on Jira.
> 
> `MentorIntake` model should be removed. Now that `Mentor` is fully updated to match the new architecture, `MentorIntake` is obsolete and it's safe to remove it.

This is correct, we are working on [#448](https://bloomtechlabs.atlassian.net/jira/software/c/projects/BL/boards/10?modal=detail&selectedIssue=BL-488&search=488)

**Review by ad-victoriam-101 — APPROVED** [review]
Looks good,

**Review by jinjahninjah — APPROVED** [review]
:shipit:

**Review by rogerfvieira — APPROVED** [review]
Great effort by all the teams on cleaning up the schema for database.

**Pull request merged**